pkg/operator/operator.go

```diff
@@ -0,0 +1,177 @@
+	// return the updated resource. All client consumers will need audited to
+	// ensure they are tolerant of stale data (or we need a cache or client that
+	// makes stronger coherence guarantees).
+	// https://pkg.go.dev/sigs.k8s.io/controller-runtime#hdr-Clients_and_Caches
+	newNoCacheClientFunc := func(_ cache.Cache, config *rest.Config, options client.Options, _ ...client.Object) (client.Client, error) {
+		return client.New(config, options)
+	}
+
+	mgr, err := ctrl.NewManager(config, ctrl.Options{
+		Logger:                 ctrl.Log.WithName("operator manager"),
+		Scheme:                 GetOperatorScheme(),
+		MetricsBindAddress:     opCfg.MetricsBindAddress,
+		Port:                   9443,
+		HealthProbeBindAddress: opCfg.HealthProbeBindAddress,
+		LeaderElection:         opCfg.EnableLeaderElection,
+		LeaderElectionID:       "94c735b6.olm.openshift.io",
+		Namespace:              opCfg.OperatorNamespace,
+		NewClient:              newNoCacheClientFunc,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create manager: %w", err)
+	}
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		return nil, fmt.Errorf("failed to set up health check: %w", err)
+
+	}
+	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
+		return nil, fmt.Errorf("failed to set up ready check: %w", err)
+	}
+
+	if err := (&nodeobservabilitycontroller.NodeObservabilityReconciler{
+		Client:     mgr.GetClient(),
+		Scheme:     GetOperatorScheme(),
+		Log:        ctrl.Log.WithName("controller.nodeobservability"),
+		Namespace:  opCfg.OperatorNamespace,
+		AgentImage: opCfg.AgentImage,
+	}).SetupWithManager(mgr); err != nil {
+		return nil, fmt.Errorf("failed to create nodeobservability controller: %w", err)
+	}
+	// Cluster is a runnable managed by controller-runtime's manager
+	// with its own client and cache.
+	// We deliberately use a dedicated cache for the watches of ca-configmap-controller
+	// to avoid the unnecessary cluster level permissions.
+	// ca-configmap-controller needs a namespace different from the operator's (source namespace)
+	// which forces the usage of the multinamespace shared cache of the manager.
+	// The additional namespace in turn obliges all the other controllers to have the rights
+	// to watch in it which results into cluster level permissions for all the operands:
+	// daemonset, serviceaccounts, services, since the local role cannot be created in another namespace via OLM.
+	// Inspired by https://github.com/kubernetes-sigs/controller-runtime/blob/master/designs/move-cluster-specific-code-out-of-manager.md
+	cluster, err := cluster.New(config, func(opts *cluster.Options) {
+		opts.NewClient = newNoCacheClientFunc
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create controller-runtime cluster: %w", err)
+	}
+
+	if err := mgr.Add(cluster); err != nil {
+		return nil, fmt.Errorf("failed to add controller-runtime cluster to manager: %w", err)
+	}
```

Comment:
Sorry didn't notice that before but this is not needed in this PR, it's needed in the next PR when ca-configmap-controller will be added. It's better to remove it from this PR to avoid any side effects.

Reply:
can you please mention the exact piece of code? I believe you are talking about this
`// Cluster is a runnable managed by controller-runtime's manager
	// with its own client and cache.
	// We deliberately use a dedicated cache for the watches of ca-configmap-controller
	// to avoid the unnecessary cluster level permissions.
	// ca-configmap-controller needs a namespace different from the operator's (source namespace)
	// which forces the usage of the multinamespace shared cache of the manager.
	// The additional namespace in turn obliges all the other controllers to have the rights
	// to watch in it which results into cluster level permissions for all the operands:
	// daemonset, serviceaccounts, services, since the local role cannot be created in another namespace via OLM.
	// Inspired by https://github.com/kubernetes-sigs/controller-runtime/blob/master/designs/move-cluster-specific-code-out-of-manager.md
	cluster, err := cluster.New(config, func(opts *cluster.Options) {
		opts.NewClient = newNoCacheClientFunc
	})`

Reply:
```// Cluster is a runnable managed by controller-runtime's manager
	// with its own client and cache.
	// We deliberately use a dedicated cache for the watches of ca-configmap-controller
	// to avoid the unnecessary cluster level permissions.
	// ca-configmap-controller needs a namespace different from the operator's (source namespace)
	// which forces the usage of the multinamespace shared cache of the manager.
	// The additional namespace in turn obliges all the other controllers to have the rights
	// to watch in it which results into cluster level permissions for all the operands:
	// daemonset, serviceaccounts, services, since the local role cannot be created in another namespace via OLM.
	// Inspired by https://github.com/kubernetes-sigs/controller-runtime/blob/master/designs/move-cluster-specific-code-out-of-manager.md
	cluster, err := cluster.New(config, func(opts *cluster.Options) {
		opts.NewClient = newNoCacheClientFunc
	})
	if err != nil {
		return nil, fmt.Errorf("failed to create controller-runtime cluster: %w", err)
	}
```